docs/source/_ext/cppkokkos.py

```diff
@@ -0,0 +1,7869 @@
+"""The C++ language domain."""
```

Comment:
@JBludau I took C++ domain and changed it to `cppkokkos` domain. This is adding stuff and changed behavior of existing stuff. This was the only way to do that. I don't expect anyone to review that extension.

Reply:
I will post a diff tomorrow morning CET.
I will check tomorrow as well compatibility with latest Sphinx. We use 5.0.1 (was the last version when we started our work) and the latest is 5.3
IMHO in Sphinx nomenclature this is just a `Domain` and it should not change in the future, maybe it could in case of some bugs. But it does not have to do much/anything with the Sphinx core.
The story is that I don't like duplicate code especially nearly 8k lines, but that was the only way we have a domain which works like C++ domain, but is customizable. One can not register twice domain or lexer with the same name. CppLexer is imported (not modified in any way), so here we will always be up to date. Back to the story I started slowly with smaller pieces, but I quickly figure out I need all of it.
Summarizing `CPPKokkosDomain` inherits from Sphinx's `Domain`, so in worst case scenario they could change the API in Sphinx 6.x.x, but I really doubt they will, because it would mean they have to change all the domains defined in Sphinx package + all user(customers) defined domains would need to be changed.

Reply:
guess we are kind of stuck with copying the 7k lines then ...

Reply:
There is OFC option to reinvent the wheel :)